x/httpproxy/proxy_handler.go

```diff
@@ -33,6 +35,14 @@ func (h *proxyHandler) ServeHTTP(proxyResp http.ResponseWriter, proxyReq *http.R
 		return
 	}
 	if proxyReq.URL.Host != "" {
+
+		pathReqUrl, err := url.Parse(proxyReq.URL.Path)
```

Comment:
I believe this will be cleaner if it's a fallback. Place it where the "Not Found" is. That's the local webserver behavior.

Then dispatcher logic is:

1. If method == CONNECT, then connect handler
2. If absolute URL (has host in URL), then forward handler
3. Fallback to webserver behavior. Map path to URL and request it.

For readability, 2 and 3 shouldn't be conflated.